ext/hash/tests/hash_serialize_001.phpt

```diff
@@ -0,0 +1,370 @@
+--TEST--
+Hash: serialize()/unserialize()
+--FILE--
+<?php
+
+$algos = hash_algos();
+
+foreach ($algos as $algo) {
+    var_dump($algo);
+    $ctx0 = hash_init($algo);
+    $serial = serialize($ctx0);
```

Comment:
Output the serialized string.  We want to make sure the expect which we get on LE arches match what we get on BE arches.

Reply:
Can we do this for a selection of algorithms and not all of them? In particular if fast-SHA3 is enabled, the serialization is still endian-sensitive.

Reply:
The new hash_serialize_003.phpt covers this case.